BraveRewardsUI/Common/DisclaimerLinkLabel.swift

```diff
@@ -0,0 +1,134 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+/// A Disclaimer Label that allows clickable links (or data-detectors)
+final class DisclaimerLinkLabel: UITextView {
+  
+  /// Called when a link is tapped
+  var onLinkedTapped: ((Link) -> Void)?
+  
+  /// Types of links supported by this label
+  public enum Link {
+    case unknown
+    case termsOfService
+    case privacyPolicy
+  }
+  
+  override var text: String? {
+    didSet {
+      updateText()
+    }
+  }
+  
+  /// Converts the text into attributed text for display
+  func updateText() {
+    guard let text = text else {
+      super.text = nil
+      return
+    }
+    
+    let attributedString = { () -> NSAttributedString? in
+      let paragraphStyle = NSMutableParagraphStyle()
+      paragraphStyle.alignment = .center
+      
+      let attributes: [NSAttributedString.Key: Any] = [.font: self.font ?? UIFont.systemFont(ofSize: 12.0),
+                                                       .foregroundColor: self.textColor ?? UX.textColor,
+                                                       .paragraphStyle: paragraphStyle]
+      
+      let text = try? NSMutableAttributedString(data: text.data(using: .utf8) ?? Data(),
+                                                          options: [.documentType: NSAttributedString.DocumentType.html,
+                                                                    .defaultAttributes: attributes],
+                                                          documentAttributes: nil)
+      
+      let range = NSRange(location: 0, length: text?.length ?? 0)
+      text?.addAttribute(.foregroundColor, value: self.textColor ?? UX.textColor, range: range)
+      text?.addAttribute(.paragraphStyle, value: paragraphStyle, range: range)
```

Comment:
Wouldn't it make sense to add these 2 lines in the `beginEditing`/`endEditing` block?

Reply:
I got rid of those lines and just passed in a dictionary and also placed it between the begin and end.